pallets/gear/rpc/src/lib.rs

```diff
@@ -214,13 +237,26 @@ where
                     allow_other_panics,
                     min_limit,
                 )
+            } else if api_version < 3 {
+                #[allow(deprecated)]
+                api.calculate_gas_info_before_version_3(
+                    at_hash,
+                    source,
+                    kind,
+                    payload,
+                    value,
+                    allow_other_panics,
+                    min_limit,
+                    Some(self.allowance_multiplier),
+                )
             } else {
                 api.calculate_gas_info(
                     at_hash,
                     source,
                     kind,
                     payload,
                     value,
+                    is_prepaid,
```

Comment:
it's wrong approach. try to pass calls maybe. It was discussed before

Reply:
added VoucherCall enum to pass msg_id or source id